### PR TITLE
1394 doc eth get block receipts api method

### PR DESCRIPTION
**Pull request by mjsmike62**
Documented the `eth_getBlockReceipts` API method in the Besu API methods/ETH methods section of the public Besu documentation.

Fixes: #1394 

See: https://besu-docs-aploqe3ms-hyperledger.vercel.app/development/public-networks/reference/api#eth_getblockreceipts

### CODE REVIEW

**Comment by matthew1001** [comment]
This looks good to me @mjsmike62, thanks

File: docs/public-networks/reference/api/index.md [hunk 1]
```diff
@@ -2775,6 +2775,194 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
 
 <!--/tabs-->
 
+### `eth_getBlockReceipts`
+
+This JSON-RPC method returns all transaction receipts for a given block and takes the block identifier as a parameter. Transaction receipts are useful because they provide a way to track the success or failure of a transaction (`1` if successful and `0` if failed), as well as the amount of gas used and any event logs that might have been produced by a smart contract during the transaction.
+
+Transaction receipts also contain the transaction hash, the block number and hash in which the transaction was included, and the contract address of any contract created as a result of the transaction.
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:2782]:
```suggestion
Returns all transaction receipts for a given block. Transaction receipts provide a way to track the success or failure of a transaction (`1` if successful and `0` if failed), as well as the amount of gas used and any event logs that might have been produced by a smart contract during the transaction.
```

If you look at other API method descriptions on this reference page, we typically keep them short. Developers looking at this page will already have an idea of what transaction receipts are, and they can learn exactly what they contain by clicking through to the linked transaction objects reference.

File: docs/public-networks/reference/api/index.md [hunk 2]
```diff
@@ -2775,6 +2775,194 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+`result`: _object_ - [block object](objects.md#block-object), or `null` when there is no block.
+
+<!--tabs-->
+
+# curl HTTP
+
+```bash
+curl -X POST "CHAINSTACK_ARCHIVE_NODE_URL" \
+  -H 'Content-Type: application/json' \
+  --data '{"method":"eth_getBlockReceipts","params":["latest"], "jsonrpc":"2.0","id":1}'
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:2801]:
```suggestion
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockReceipts","params":["latest"],"id":1}' http://127.0.0.1:8545
```

Update to match the curl request examples for the other API methods on this reference page. Currently, they're all on one line and use the local RPC URL (`http://127.0.0.1:8545`).

**Review by alexandratran — COMMENTED** [review]
Added two change suggestions. Again, you can go to **Files changed**, and for each suggestion, **Add suggestion to batch**, then commit the suggestions directly in GH.

File: docs/public-networks/reference/api/index.md [hunk 3]
```diff
@@ -2775,6 +2775,194 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+    }
+    miner {
+      address
+    }
+    account(address: "0xd42e2b1c14d02f1df5369a9827cb8e6f3f75f338") {
+      balance
+    }
+    parent {
+      hash
+    }
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2919]: This data is relevant to the block rather than the transaction receipt.

I think it's better to base off the example in https://besu-docs-60seh79zu-hyperledger.vercel.app/development/public-networks/reference/api#eth_gettransactionreceipt

e.g.
```
{ 
  block  (hash: "0x0ed9da11f2b4afdf19710402c1891fd23b9e7494a7c38036832d9db1296e846c") {
    transactions {
      block {
        hash 
        logsBloom
      } 
      hash 
      createdContract { 
        address 
      } 
      cumulativeGasUsed
      gas
      gasUsed
      logs{ 
        topics
      } 
      from {
        address
      } 
      to {
        address
      } 
      index
    }
  }
}  
```

File: docs/public-networks/reference/api/index.md [hunk 4]
```diff
@@ -2775,6 +2775,194 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+  ]
+}
+```
+
+# curl GraphQL
+
+```bash
+curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (number : 100) {transactions{hash} timestamp difficulty totalDifficulty gasUsed gasLimit hash nonce ommerCount logsBloom mixHash ommerHash extraData stateRoot receiptsRoot transactionCount transactionsRoot ommers{hash} ommerAt(index : 1){hash} miner{address} account(address: \"0xd42e2b1c14d02f1df5369a9827cb8e6f3f75f338\"){balance} parent{hash} }}"}' http://localhost:8547/graphql
+```
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2879]: Same point as for the GraphQL one

Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2879]: For curl GraphQL (lines 2878-2879), the code is already on a single line as seen in the output (http://localhost:3000/development/public-networks/reference/api#eth_getblockreceipts).  Simon, is this format okay with you?

Comment (siladu) [reply docs/public-networks/reference/api/index.md:2875]: Sorry, yes this single line format is OK, I meant the contents of the graphql request in this curl one is relevant for block rather than transaction receipts. I think it should match the non-curl GraphQL query

File: docs/public-networks/reference/api/index.md [hunk 5]
```diff
@@ -2775,6 +2775,177 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+
+# GraphQL
+
+```text
+{
+  transaction(hash: "0x5f5366af89e8777d5ae62a1af94a0876bdccbc22417bed0aff361eefa3e37f86") {
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2881]: Think this may be straight copy of the getTransactionReceipt request?

getBlockReceipts needs to pass the block hash/number/tag in as a param, then return the list of transactionReceipts, so it's a combination of what you had before and this.

File: docs/public-networks/reference/cli/options.md [hunk 6]
```diff
@@ -1167,6 +1167,38 @@ The key-value storage to use. Use this option only if using a storage system pro
 
 For development use only, the `memory` option provides ephemeral storage for sync testing and debugging.
 
+### `--kzg-trusted-setup`
```
Comment (siladu) [inline docs/public-networks/reference/cli/options.md:1170]: Think this has come in from unrelated commit?

Comment (mjsmike62) [reply docs/public-networks/reference/cli/options.md:1170]: Oops, yes. This should not have been included. Don't know how that happened since I created a separate topic branch for issue [Document new CLI option #1274](https://github.com/hyperledger/besu-docs/issues/1274).

File: docs/public-networks/reference/api/index.md (hunk 4 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md]: Corrected as per the source you supplied me in Slack.

File: docs/public-networks/reference/cli/options.md (hunk 6 above)
Comment (mjsmike62) [reply docs/public-networks/reference/cli/options.md]: Removed the `kzg-trusted-setup` CL option from the topic branch since it was already merged into main in a previous issue.

File: docs/public-networks/reference/api/index.md (hunk 5 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md]: This code has been replaced with the correct code as per the source that you provided.

File: docs/public-networks/reference/api/index.md (hunk 2 above)
Comment (siladu) [reply docs/public-networks/reference/api/index.md:2800]: @mjsmike62 Looks like this change has reverted back to before the suggestion.

File: docs/public-networks/reference/api/index.md [hunk 7]
```diff
@@ -2775,6 +2775,179 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+```
+
+# GraphQL result
+
+```json
+{
+  "data": {
+    "block": {
+      "transactions": [],
+      "timestamp": "0x5cd10933",
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2916]: This isn't quite right: this json is showing the result of a block with 0 transactions and is showing _all_ the block's details even though we've only requested the transaction details. The shape of the json result should look more like this...

```
{
  "data" : {
    "block" : {
      "transactions" : [ {
        "block" : {
          "hash" : "0x4d746a3381673a5180744a56e78cded4696b77317866c2253566e0fa16967e1d",
          "logsBloom" : "0x2e0a8080520608000e38181e0c9081e813a00c184a010d1900c9602240428dc6480004444098428b945010802454104002827420426591a200224016802841900031bd4440828ec9b113081880027c01cc47105c1885d556216200880026160810050028422a4b0c4bc8087372860851000802c8d901158504a482100d488040119c08045e500824402054a0d91cc433188909020a06ac841914a2a082c104a1260460014b8b001b28030202518c040008266038a880026208041d082503589054581223c188004396804801280c00020c492816060a421831c8820ac04460303a9e48128238e0098f319030083808150c4914b8840000206715481500690000"
        },
        "hash" : "0x7afe779fd0c6d4a1b6f330e679a5cf94095eaa57d2ce0c0ef991dfb2b405374f",
        "createdContract" : null,
        "cumulativeGasUsed" : "0x5208",
        "gas" : "0x61a8",
        "gasUsed" : "0x5208",
        "logs" : [ ],
        "from" : {
          "address" : "0x66f962241b8ff853849c85a63a0ce20bae4f68d5"
        },
        "to" : {
          "address" : "0x6be8356826a9fc7b2d911fcc1de6342ae5f5b9a3"
        },
        "index" : "0x0"
      }, {
        "block" : {
          "hash" : "0x4d746a3381673a5180744a56e78cded4696b77317866c2253566e0fa16967e1d",
          "logsBloom" : "0x2e0a8080520608000e38181e0c9081e813a00c184a010d1900c9602240428dc6480004444098428b945010802454104002827420426591a200224016802841900031bd4440828ec9b113081880027c01cc47105c1885d556216200880026160810050028422a4b0c4bc8087372860851000802c8d901158504a482100d488040119c08045e500824402054a0d91cc433188909020a06ac841914a2a082c104a1260460014b8b001b28030202518c040008266038a880026208041d082503589054581223c188004396804801280c00020c492816060a421831c8820ac04460303a9e48128238e0098f319030083808150c4914b8840000206715481500690000"
        },
        "hash" : "0x412f04ba27c1c096dadb2d8af54ee61034c3d4679fdd025a634e95fa2238713c",
        "createdContract" : null,
        "cumulativeGasUsed" : "0xbcdb2",
        "gas" : "0xbdfe0",
        "gasUsed" : "0xb7baa",
        "logs" : [ {
          "topics" : [ "0xd93fde3ea1bb11dcd7a4e66320a05fc5aa63983b6447eff660084c4b1b1b499b", "0x00000000000000000000000000000000000000000000000000000000000e4d3a" ]
        } ],
        "from" : {
          "address" : "0xe253f7a6533c62755f470b33fa5bcd659a5db3cd"
        },
        "to" : {
          "address" : "0x95ff8d3ce9dcb7455beb7845143bea84fe5c4f6f"
        },
        "index" : "0x1"
      } ]
    }
  }
}
```

File: docs/public-networks/reference/api/index.md [hunk 8]
```diff
@@ -2775,6 +2775,179 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
 <!--/tabs-->
 
+### `eth_getBlockReceipts`
+
+Returns all transaction receipts for a given block. Transaction receipts provide a way to track the success or failure of a transaction (`1` if successful and `0` if failed), as well as the amount of
+gas used and any event logs that might have been produced by a smart contract during the transaction.
+
+#### Parameters
+
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2785]:
```suggestion
- `blockNumber` or `blockHash`: _string_ - hexadecimal or decimal integer representing a block number, block hash, or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/public-networks/reference/api/index.md [hunk 9]
```diff
@@ -2775,6 +2775,179 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+
+- `verbose`: _boolean_ - if `true`, returns the full [transaction objects](objects.md#transaction-object); if `false`, returns only the hashes of the transactions.
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2787]: `verbose` is not one of the parameters for this method, it's just the block parameter: https://github.com/ethereum/execution-apis/blob/057892b256c181232ca22499d1801b9637192698/src/eth/block.yaml#L89-L95

**Review by siladu — CHANGES_REQUESTED** [review]
Found a few issues. Also, for some reason I can't see eth_getBlockReceipts in the latest preview: https://besu-docs-pck4y18td-hyperledger.vercel.app/public-networks/reference/api

File: docs/public-networks/reference/api/index.md (hunk 7 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2916]: Simon, thanks for the explanation. Your provided code was incorporated.

File: docs/public-networks/reference/api/index.md (hunk 8 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2785]: Edit incorporated.

File: docs/public-networks/reference/api/index.md (hunk 9 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2787]: Block parameter has been incorporated.

File: docs/public-networks/reference/api/index.md [hunk 10]
```diff
@@ -2775,6 +2775,192 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+gas used and any event logs that might have been produced by a smart contract during the transaction.
+
+#### Parameters
+
+- `blockNumber` or `blockHash`: _string_ - hexadecimal or decimal integer representing a block number, block hash, or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
+
+- name: Block
+      required: true
+      schema:
+        $ref: '#/components/schemas/BlockNumberOrTagOrHash'
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2790]: I think I have confused you here, this whole second parameter should be removed. This is part of the ethereum api spec, not appropriate for these besu docs, I was just linking to it so you could see that there should only be one parameter

File: docs/public-networks/reference/api/index.md [hunk 11]
```diff
@@ -2775,6 +2775,192 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+
+```bash
+curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block  (hash: \"0x0ed9da11f2b4afdf19710402c1891fd23b9e7494a7c38036832d9db1296e846c\") {transactions{block{hash logsBloom} hash createdContract{address} cumulativeGasUsed gas gasUsed logs{topics} from{address} to{address} index}}}"}' http://localhost:8547/graphql
+```
+
+# GraphQL
+
+```text
+{
+  block (hash: "0x0ed9da11f2b4afdf19710402c1891fd23b9e7494a7c38036832d9db1296e846c") {
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2884]: nit: since you've updated the result, it would be nice to have the requests' block hash match the result's block hash 
```suggestion
  block (hash: "0x4d746a3381673a5180744a56e78cded4696b77317866c2253566e0fa16967e1d") {
```

File: docs/public-networks/reference/api/index.md [hunk 12]
```diff
@@ -2775,6 +2775,192 @@ curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block (num
...
+      "type": "0x2"
+    }
+  ]
+}
+```
+
+# curl GraphQL
+
+```bash
+curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block  (hash: \"0x0ed9da11f2b4afdf19710402c1891fd23b9e7494a7c38036832d9db1296e846c\") {transactions{block{hash logsBloom} hash createdContract{address} cumulativeGasUsed gas gasUsed logs{topics} from{address} to{address} index}}}"}' http://localhost:8547/graphql
```
Comment (siladu) [inline docs/public-networks/reference/api/index.md:2877]: nit: since you've updated the result, it would be nice to have the requests' block hash match the result's block hash

```suggestion
curl -X POST -H "Content-Type: application/json" --data '{ "query": "{block  (hash: \"0x4d746a3381673a5180744a56e78cded4696b77317866c2253566e0fa16967e1d\") {transactions{block{hash logsBloom} hash createdContract{address} cumulativeGasUsed gas gasUsed logs{topics} from{address} to{address} index}}}"}' http://localhost:8547/graphql
```

**Review by siladu — COMMENTED** [review]
Couple more thing. This is the reverted issue I mentioned: https://github.com/hyperledger/besu-docs/pull/1413/files#r1376752181

File: docs/public-networks/reference/api/index.md (hunk 10 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2790]: Understand. Second parameter has been removed.

File: docs/public-networks/reference/api/index.md (hunk 11 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2884]: Understand. Good catch. Edit incorporated.

File: docs/public-networks/reference/api/index.md (hunk 12 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2877]: Understand. Good catch. Edit incorporated.

File: docs/public-networks/reference/api/index.md (hunk 2 above)
Comment (mjsmike62) [reply docs/public-networks/reference/api/index.md:2798]: Edit has been incorporated.

**Review by siladu — APPROVED** [review]
LGTM!

**Review by alexandratran — APPROVED** [review]
lgtm
